# code

In [1]:
import sys


class atm():
    def __init__(self,pin1 , balance=0):
        self.pin=pin1
        self.balance = balance


    def deposit(self, amount):                      # to deposit amount in the account
        self.amount = amount
        self.balance = self.balance + self.amount
        print("Current account balance: ", self.balance)
        print()

    def withdraw(self, amount):                         # for withdrawing the amount
        self.amount = amount
        if self.amount > self.balance:
            print("Insufficient fund!")
            print(f"Your balance is {self.balance} only.")
            print("Try with lesser amount than balance.")
            print()
        else:
            self.balance = self.balance - self.amount
            print(f"Nu.{amount} withdrawal successful!")
            print("Current account balance: ", self.balance)
            print()

    def transferfunds(self, amount):                    # to transfer funds from checking balance to saving balance
        self.amount = amount
        checking_balance = 10000-self.amount

        self.balance = self.balance +self.amount
        print("Account balance: ", self.balance)
        print()
        print("Checking Balance: ", checking_balance)
        print()



    def check_balance(self):                           # to get balance in the account
        print("Available balance: ", self.balance)
        print()

    def transaction(self):

        while True:                                   # loop to get options
            try:
                option = int(input("Enter 1, 2, 3, 4 "))
            except:
                print("Error: Enter 1, 2, 3, 4 only!!!\n")
                continue
            else:
                if option == 1:
                    atm.check_balance()
                elif option == 2:
                    amount = int(input("How much you want to deposit:"))
                    atm.deposit(amount)
                elif option == 3:
                    amount = int(input("How much you want to withdraw:"))
                    atm.withdraw(amount)
                elif option == 4:
                    amount = int(input("Enter the amount to transfer:"))
                    atm.transferfunds(amount)
                else:
                    print("Wrong input")
                    sys.exit()




user={'1234':0.0}         # to save user pin and the balance in that user account
print("*******WELCOME TO ATM*******")
print("___________________________________________________________\n")
pin1 = input("Enter the pin: ")
pin2=input("Enter the pin again: ")
if pin1 != pin2:
    print("Wrong pin. Try again.")
elif pin1 not in user:
    print("Not a registered user")
    x=input("To register enter R else N")
    if x=='R':
        data=input("Enter the pin")
        if data in user:
            print("Enter another pin")
        else:
            user[data]=0.0
    elif x=='N':
        sys.exit()
    else:
        print("Invalid input")
else:
    print("Pin verified")

atm=atm(pin1)


while True:             # id the user want to perform any kind of transaction in their account
    trans = input("Do you want to do any transaction?(y/n):")
    if trans == "y":
        atm.transaction()
    elif trans == "n":
        print("Thank you")
        break
    else:
        print("Wrong command!\n")


*******WELCOME TO ATM*******
___________________________________________________________

Enter the pin: 1234
Enter the pin again: 1234
Pin verified
Do you want to do any transaction?(y/n):
Wrong command!

Do you want to do any transaction?(y/n):y
Enter 1, 2, 3, 4 1
Available balance:  0

Enter 1, 2, 3, 4 2
How much you want to deposit:100
Current account balance:  100

Enter 1, 2, 3, 4 3
How much you want to withdraw:20
Nu.20 withdrawal successful!
Current account balance:  80

Enter 1, 2, 3, 4 4
Enter the amount to transfer:10
Account balance:  90

Checking Balance:  9990

Enter 1, 2, 3, 4 5
Wrong input


SystemExit: 

D:\project\lib\site-packages\IPython\core\interactiveshell.py:3377: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)


# output

*******WELCOME TO ATM*******
___________________________________________________________

Enter the pin: 1234
Enter the pin again: 1234
Pin verified
Do you want to do any transaction?(y/n):
Wrong command!

Do you want to do any transaction?(y/n):y
Enter 1, 2, 3, 4 1
Available balance:  0

Enter 1, 2, 3, 4 2
How much you want to deposit:100
Current account balance:  100

Enter 1, 2, 3, 4 3
How much you want to withdraw:20
Nu.20 withdrawal successful!
Current account balance:  80

Enter 1, 2, 3, 4 4
Enter the amount to transfer:10
Account balance:  90

Checking Balance:  9990

Enter 1, 2, 3, 4 5
Wrong input